ynr/apps/sopn_parsing/helpers/pdf_helpers.py

```diff
@@ -102,27 +170,82 @@ def get_page_heading(self):
+        search_text = self.get_page_heading()
+        wards = ward.split("/")
+        for ward in wards:
+            if ward in search_text:
+                self.matched_post_label = ward
+                return True
+        return False
+
+    def set_continuation_page(self, value, previous_page=None):
+        self.continuation_page = value
+        if value:
+            if not self.matched_post_label:
+                self.matched_post_label = previous_page.matched_post_label
+
+        return self.continuation_page
+
+    def is_continuation_page(self, document_heading, previous_page, post_label):
         """
         Take a set containing the document heading (returned from
         `get_page_heading_set`) and compare it to another heading set.
-
         This is done by taking the intersection of the two sets. If the length
         of the intersection set divided by the length of the provided
         document_heading set is less than CONTINUATION_THRESHOLD then we assume
-        this is a "continuation" page and return False.
-
+        this is a "continuation" page and return True.
         If the divided number is greater than the CONTINUATION_THRESHOLD then we
-        assume this is a top page and return True.
-
+        assume this is a top page and return False.
+        If the headings up to the post label are identical, we assume the page
+        is a continuation.
         """
+        post_label = clean_text(post_label)
+
+        # CASE 1: We know the first page is never a continuation page.
+        if self.page_number == 1:
+            return self.set_continuation_page(False)
+
+        # CASE 2: If the document heading is very different to the document
+        # heading then we assume this is a continuation page
         similar_len = document_heading.intersection(self.get_page_heading_set())
-        if len(similar_len) / len(document_heading) < CONTINUATION_THRESHOLD:
-            self.is_top_page = False
-        return self.is_top_page
+        is_very_different_to_doc_heading = (
+            len(similar_len) / len(document_heading) < CONTINUATION_THRESHOLD
+        )
+        if is_very_different_to_doc_heading:
+            return self.set_continuation_page(True, previous_page)
+
+        # CASE 3: If the headings are more or less the same, split on the ward
+        # name and see if they're identical
+        previous_page_heading = previous_page.get_page_heading()
+        previous_page_heading_up_to_ward_name = " ".join(
+            previous_page_heading.partition(post_label)[0:2]
+        )
+
+        page_heading = self.get_page_heading()
+        page_heading_up_to_ward_name = " ".join(
+            page_heading.partition(previous_page.matched_post_label)[0:2]
+        )
+
+        headings_are_identical = (
+            page_heading_up_to_ward_name
+            == previous_page_heading_up_to_ward_name
+        )
+
+        # TO DO: multipage sopns with multiple wards where headings
```

Comment:
Isn't what you describe the correct behaviour in this case? The pages are for different ballots, so `is_continuation_page` shouldn't return True (we might want it to always return False rather than None, but that's not the issue here).

Reply:
Yes I agree - went through this with Virginia and cleared up the issues she was having (because it was a by-election SOPN all pages were being returned incorrectly)